### DISCUSSION TITLE
How to assemble partials without rendering variables

### DISCUSSION THREAD

**Discussion by yespire** [handlebars-lang/handlebars.js#1798] [Q&A]
New to handlebars, so what I am asking may be already exists:

Is there a way to assemble the partials without rendering variables ?

```
h.registerPartial("p1", "p1")
h.registerPartial("p2", "p2")
let output = h.compile("assemble partials: {{> p1}} + {{> p2}} without rendering variables: {{v1}} {{v2}}")()
```
right now, this produces "`assemble partials: p1 + p2 without rendering variables: `"

is there a way to produce "`assemble partials: p1 + p2 without rendering variables: {{v1}} {{v2}}`"

I know putting \ before {{ to make rendering skip it, but this is NOT as good as IF there is a function just assemble partials

-----

Possible Syntax 1 - new method:
~ this is preferred
```
h.assemblePartials("assemble partials: {{> p1}} + {{> p2}} without rendering variables: {{v1}} {{v2}}")
```
Possible Syntax 2 - new option for compile: 
~ this doesn't feel right if rendering call includes any template variable
```
h.compile("assemble partials: {{> p1}} + {{> p2}} without rendering variables: {{v1}} {{v2}}", {
   onlyAssemblePartials:true
})()
```

**Comment by jaylinski** [comment]
@yespire As a workaround, you can replace your variables with variables , something like this:

https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%3Epartial%7D%7D%5Cn%7B%7Bfirstname%7D%7D%20%7B%7Blastname%7D%7D%5Cn%22%2C%22partials%22%3A%5B%7B%22name%22%3A%22partial%22%2C%22content%22%3A%22Partial%3A%5Cn%22%7D%5D%2C%22input%22%3A%22%7B%5Cn%20%20firstname%3A%20%5C%22%7B%7B%20firstname%20%7D%7D%5C%22%2C%5Cn%20%20lastname%3A%20%5C%22%7B%7B%20lastname%20%7D%7D%5C%22%2C%5Cn%7D%5Cn%22%2C%22output%22%3A%22Partial%3A%5Cn%7B%7B%20firstname%20%7D%7D%20%7B%7B%20lastname%20%7D%7D%5Cn%22%2C%22preparationScript%22%3A%22%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D

When the partials are assembled, you can provide the actual data for those variables.

If you still want to add something like `h.assemblePartials()`, please open another issue or a PR.

**Comment by yespire** [comment]
@jaylinski 

Thanks for the suggestion, but rendering variables into variables or \{{varirable}} are not ideal. 

Are you suggesting create a "feature" issue ?

**Reply by jaylinski** [reply]
> Are you suggesting create a "feature" issue ?

yes

**Reply by yespire** [reply]
Created

**Comment by yespire** [comment]
Cool, will do.

assemblePartials() can solve some of the other request(s) / discussion(s), for example:
https://github.com/handlebars-lang/handlebars.js/discussions/1788
